app/services/customers/create_service.rb

```diff
@@ -57,7 +57,7 @@ def create_from_api(organization:, params:)
         customer.lastname = params[:lastname] if params.key?(:lastname)
         customer.customer_type = params[:customer_type] if params.key?(:customer_type)
 
-        if customer.organization.revenue_share_enabled?
+        if customer.organization.revenue_share_enabled? && customer.editable?
```

Comment:
I would also include the "editable?" check in the other public method `create` around line 178 of this same file, and also cover it with tests...